contributors/design-proposals/flakiness-sla.md

```diff
@@ -0,0 +1,92 @@
+# Kubernetes Testing Flakiness SLA
+
+This document captures the expectations of the community about flakiness in
+our tests and our test infrastructure. It sets out an SLA (Service Level 
+Agreement) for flakiness in our tests, as well as actions that we will
+take when we are out of SLA.
+
+
+## Definition of a "Flake"
+
+We'll start by the definition of a _flake_. _Flakiness_ is defined for a
+complete run of one required job in the pre-submit testing infrastructure 
+(e.g. pull-kubernetes-e2e-gke-gci) for a given pull request (PR). We will
+not measure flakiness SLA for e2e jobs that are not required for code merge.
+
+A pre-submit job's test result is considered to
+be a flake according to two criteria:
+
+1) it both fails and passes without any changes in the code
```

Comment:
We already detect this with the ginkgo tests on presubmits, there's no need to actually run the test again. (If you have to run the test a second time, it's actually failing 3/4 times, not 1/2 times)